akka-remote/src/main/scala/akka/remote/serialization/MiscMessageSerializer.scala

```diff
@@ -129,6 +173,34 @@ class MiscMessageSerializer(val system: ExtendedActorSystem) extends SerializerW
     }
   }
 
+  private def deserializeStatusSuccess(bytes: Array[Byte]): Status.Success =
+    Status.Success(deserializePayload(ContainerFormats.Payload.parseFrom(bytes)))
+
+  private def deserializeStatusFailure(bytes: Array[Byte]): Status.Failure =
+    Status.Failure(deserializePayload(ContainerFormats.Payload.parseFrom(bytes)).asInstanceOf[Throwable])
+
+  private def deserializeThrowable(bytes: Array[Byte]): Throwable = {
+    val protoT = ContainerFormats.Throwable.parseFrom(bytes)
+    val t: Throwable =
+      if (protoT.hasCause) {
+        val cause = deserializePayload(protoT.getCause).asInstanceOf[Throwable]
+        system.dynamicAccess.createInstanceFor[Throwable](
+          protoT.getClazzName,
+          List(classOf[String] → protoT.getMessage, classOf[Throwable] → cause)).get
+      } else
+        system.dynamicAccess.createInstanceFor[Throwable](
+          protoT.getClazzName,
+          List(classOf[String] → protoT.getMessage)).get
```

Comment:
This is safe since we know that it is a `Throwable`? A malicious user could select this deserializer through the manifest and then make it deserialize whatever class on the classpath?